### PR TITLE
Centralizing permission checks for views.

### PR DESCRIPTION
**Pull request by dennisoelkers**
## Motivation and Context

Prior to this PR, permission checks for saved searches/dashboards/searches/streams in the views code were implemented over and over again in REST resource classes, prone for failure and forgetting things.

In this PR, permission checks are centralized in a newly-introduced `SearchUser` class, which can be injected through Hk2 in a REST resource method using the `@Context` annotation. This instance can then be passed on completely to consumers which need to perform permission checks or individual methods of it can be handed over as predicates. This allows us to have a central place where we can implement and manage permission checks as well as abstracting from the actual implementation through `isPermitted`.

/jenkins-pr-deps Graylog2/graylog-plugin-enterprise#2918

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: graylog2-server/src/main/java/org/graylog/plugins/views/search/SearchDomain.java [hunk 1]
```diff
@@ -39,27 +40,27 @@ public SearchDomain(SearchDbService dbService, ViewService viewService) {
         this.viewService = viewService;
     }
 
-    public Optional<Search> getForUser(String id, User user, Predicate<ViewDTO> viewReadPermission) {
+    public Optional<Search> getForUser(String id, SearchUser searchUser) {
         final Optional<Search> search = dbService.get(id);
 
-        search.ifPresent(s -> checkPermission(user, viewReadPermission, s));
+        search.ifPresent(s -> checkPermission(searchUser.username(), searchUser, s));
```
Comment (janheise) [inline graylog2-server/src/main/java/org/graylog/plugins/views/search/SearchDomain.java:46]: checkPermission without String username + searchUser.username() to create the Exception?

**Review by janheise — COMMENTED** [review]
Looks super! found 1 little thing that might be a candidate for a small change

**Review by todvora — APPROVED** [review]
Like it a lot, looks very good! Thank you!

File: graylog2-server/src/main/java/org/graylog/plugins/views/search/SearchDomain.java (hunk 1 above)
Comment (dennisoelkers) [reply graylog2-server/src/main/java/org/graylog/plugins/views/search/SearchDomain.java]: Good idea! Changed that.

**Review by janheise — APPROVED** [review]

**Pull request merged**